### PR TITLE
Fix library focus RTE

### PR DESCRIPTION
**Pull request by JoergAtGithub**
Closes:  #12065

### CODE REVIEW

**Review by Holzhaus — APPROVED** [review]
Thanks, code looks good to me, but I'd appreciate confirmation from @ronso0.

PS: This should really use the early exit technique instead of nesting 4 `if` statements. But this is just a bugfix, so it's out of scope here.

File: src/widget/wlibrarytableview.cpp [hunk 1]
```diff
@@ -274,7 +274,7 @@ void WLibraryTableView::focusInEvent(QFocusEvent* event) {
         // This is especially helpful if the table has only one track, which can
         // not be selected with up/down buttons, either physical or emulated via
         // [Library],MoveVertical controls. See #9548
-        if (model()->rowCount() > 0) {
+        if (model() != nullptr && model()->rowCount() > 0) {
```
Comment (ronso0) [inline src/widget/wlibrarytableview.cpp:277]: Why not just `if (model() && model()->rowCount() > 0) {`?

**Comment by ronso0** [comment]
Thanks!
I suggest to squash into one commit and rephrase the message to clarify the fix (not which symptom it avoids), something along the lines of
"WLibraryTableView: assert model when selecting row"

**Comment by uklotzde** [comment]
> Thanks! I suggest to squash into one commit and rephrase the message to clarify the fix...

My recommendation for Mixxx:

image

One commit on upstream per PR with a customizable commit message on merge. Clean history with meaningful commits under your control. Example: https://github.com/tokio-rs/tokio/commits/master

**Comment by daschuer** [comment]
We have discussed the sqash merging earlier and decided against. The reason is that we loose the meaningful commit messages form the intermediate commits. We want to have them in our repository for bisect and blame. We don't want to rely on the history kept by GitHub that is not under our control.

**Comment by ronso0** [comment]
While this argmuent is true for most cases, squash & merge would be helpful in (admittedly rare) cases like this where we're talking about tiny commits and fixups that can(should) be squashed safely without loosing meaningful info.

**Comment by ronso0** [comment]
**PS**

IMO this would accelerate the workflow.
This PR would already have been merged by now, but we need the contributor to do the work locally, push, then merge.

IMO we should enable Squash & merge **but** define the conditions precisely. E.g. one-commit PRs with additional fixups, text-only changes (new blog post, adding manual chapters), new controller mappings.

Let's discuss that on Zulip later on, maybe when 2.4 is out the door?

--enough off-topic--

This is ready. @JoergAtGithub would you mind to squash & rename?

**Comment by JoergAtGithub** [comment]
I did already

**Comment by ronso0** [comment]
I'm sorry I overlooked that, Notifications send me to the comment after.

Thank you, LGTM and thanks for quickly fixing my failure, of course!

**Pull request merged**